doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -145,31 +126,55 @@ Programmable proof search
    .. seealso:: :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>`
 
 
-.. tacn:: autounfold with {+ @ident}
-   :name: autounfold
+.. tacn:: autounfold {? @hintbases } {? @clause_dft_concl }
 
-   This tactic unfolds constants that were declared through a :cmd:`Hint Unfold`
-   in the given databases.
+   .. insertprodn clause_dft_concl hypident
+
+   .. prodn::
+      clause_dft_concl ::= in @in_clause
+      | {? at @occs_nums }
+      in_clause ::= * {? at @occs_nums }
+      | * %|- {? @concl_occ }
+      | {*, @hypident_occ } {? %|- {? @concl_occ } }
+      concl_occ ::= * {? at @occs_nums }
+      hypident_occ ::= @hypident {? at @occs_nums }
+      hypident ::= @ident
+      | ( type of @ident )
+      | ( value of @ident )
 
-.. tacv:: autounfold with {+ @ident} in @goal_occurrences
+   .. todo PR: move occs_num here from the rewriting chapter?  clause_dft_concl is used a lot BTW
 
-   Performs the unfolding in the given clause (:token:`goal_occurrences`).
+   :n:`in @in_clause`
+     Performs the unfolding in the given clause.
 
-.. tacv:: autounfold with *
+   :n:`with *`
+     Use the unfolding hints in all existing hint databases.
 
-   Uses the unfold hints declared in all the hint databases.
+   :n:`with {+ @ident }`
+     Use the unfolding hints in :n:`{+ @ident }` in addition to the database ``core``.
+     Use the fake database `nocore` to omit `core`.
 
-.. tacn:: autorewrite with {+ @ident}
-   :name: autorewrite
+   Unfolds constants that were declared through a :cmd:`Hint Unfold`
+   in the given databases.
+
+.. tacn:: autorewrite {? * } with {+ @ident } {? @clause_dft_concl } {? using @ltac_expr }
```

Comment:
@mattam82 (or @JasonGross): can you tell us what this optional `*` does?

Reply:
Can you guess from this?  The `*` passes a `~conds:AllMatches` to the lower level code.  `Naive` is the default for `~conds`.

The options are defined as:

```
type conditions =
  | Naive (* Only try the first occurrence of the lemma (default) *)
  | FirstSolved (* Use the first match whose side-conditions are solved *)
  | AllMatches (* Rewrite all matches whose side-conditions are solved *)
```

Reply:
From this, I would conclude approximately that `*` means `autorewrite` will rewrite in all occurrences that match a single rewrite lemma instead of rewriting in only one place at once, and that it may do more progress in some cases because of this.

Reply:
This seems to be a bit of a mess.  Only some parts of the `occurrences` clause are accepted.  I guess the `*` is a workaround for not supporting the `at 1` form.  Would you mind turning my last example into something that works and verifying that the occurrence numbers are applied in H?

```
autorewrite * with True at 1.
(* The "at" syntax isn't available yet for the autorewrite tactic. *)
autorewrite * with True in |- * at 1.
(* The "at" syntax isn't available yet for the autorewrite tactic. *)
autorewrite * with True in H at 1.
(* Rewriting base True does not exist. *)
```

Reply:
See https://github.com/coq/coq/issues/4976 and https://github.com/coq/coq/issues/7672

Reply:
Also https://github.com/coq/coq/issues/1933#issuecomment-337497938 where @mattam82 seems to have given an explanation (in French) of autorewrite * in 2010

Reply:
@JasonGross Thanks for your help. @mattam82's French comment about `rewrite*` and `autorewrite*` aligns well with the piece of code found by Jim but the issue opened by Samuel seems to indicate that `autorewrite*` does not actually work. We need @mattam82 to comment, so I'll ping him by e-mail.

In any case, the current documentation of `*` added in this PR is probably a bit too simple ("If present, rewrite all occurrences.")